pkg/codegen/pcl/binder_schema.go

```diff
@@ -273,7 +307,13 @@ func (b *binder) loadReferencedPackageSchemas(n Node) error {
 			continue
 		}
 
-		pkg, err := b.options.packageCache.loadPackageSchema(b.options.loader, name, pkgOpts.version)
+		var pkg *packageSchema
+		var err error
+		if packageDescriptor, ok := b.packageDescriptors[name]; ok {
+			pkg, err = b.options.packageCache.loadPackageSchemaFromDescriptor(b.options.loader, packageDescriptor)
+		} else {
+			pkg, err = b.options.packageCache.loadPackageSchema(b.options.loader, name, pkgOpts.version)
+		}
```

Comment:
Feels like another option here is to construct a `packageDescriptor` manually if there isn't one in the map (with no parameterization), and then always call `loadPackageSchemaFromDescriptor` (with a view to deprecating `loadPackageSchema`). Same in the previous instance.